lib/ansible/modules/cloud/amazon/rds_snapshot.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/python
+# Copyright (c) 2014-2017 Ansible Project
+# Copyright (c) 2017, 2018 Will Thames
+# Copyright (c) 2017, 2018 Michael De La Rue
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'metadata_version': '1.1'}
+
+DOCUMENTATION = '''
+---
+module: rds_snapshot
+version_added: "2.9"
+short_description: manage Amazon RDS snapshots
+description:
+     - Creates or deletes RDS snapshots.
+options:
+  state:
+    description:
+      - Specify the desired state of the snapshot
+    default: present
+    choices: [ 'present', 'absent']
+  db_snapshot_identifier:
+    description:
+      - The snapshot to manage
+    required: true
+    aliases:
+      - id
+  db_instance_identifier:
+    description:
+      - Database instance identifier. Required when state is present
+    aliases:
+      - instance
+  wait:
+    description:
+      - Whether or not to wait for snapshot creation or deletion
+    type: bool
+    default: 'no'
+  wait_timeout:
+    description:
+      - how long before wait gives up, in seconds
+    default: 300
+  tags:
+    description:
+      - tags dict to apply to a snapshot.
+requirements:
+    - "python >= 2.6"
+    - "boto3"
+author:
+    - "Will Thames (@willthames)"
+    - "Michael De La Rue (@mikedlr)"
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Create snapshot
+- rds_snapshot:
+    instance_name: new-database
+    snapshot: new-database-snapshot
+
+# Delete snapshot
+- rds_snapshot:
+    snapshot: new-database-snapshot
+    state: absent
+
+- debug:
+    msg: "The new db endpoint is {{ rds.instance.endpoint }}"
+'''
+
+RETURN = '''
+snapshot:
+  description: the information returned in data from boto3 get_db_snapshot
+  returned: success
+  type: dict
+'''
+
+try:
+    import botocore
+except ImportError:
+    pass  # protected by AnsibleAWSModule
+
+import time
+
+# import module snippets
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.aws.rds import get_snapshot, snapshot_to_facts
+
+
+def await_resource(conn, resource_name, status, module):
+    wait_timeout = module.params.get('wait_timeout') + time.time()
+    resource = get_snapshot(conn, resource_name)
+    while not resource or (wait_timeout > time.time() and resource['Status'] != status):
```

Comment:
There are some built-in waiters that could be used instead. Besides that and the docs suggestions, looks good to me. https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/rds.html#waiters

Aurora still needs some work in general and since the rds module cannot handle aurora for snapshots either yet I don't think it should be a blocker.

Reply:
ok, that's good to hear @s-hertel